### DISCUSSION TITLE
SPARC64 emulation: QEMU error: qemu-sparc64-softmmu: Virtio VGA not available

### DISCUSSION THREAD

**Discussion by lelegard** [utmapp/UTM#6024] [UTM Help (macOS)]
Hi,

I want to setup an emulated SPARC64 VM with a Debian guest OS.

- Use UTM 4.4.5 (94) on an Intel iMac (also tried on an Arm64 MacBook M1).
- Got debian-12.0.0-sparc64-NETINST-1.iso as installation image for the guest.
- Created a new emulated virtual machine, arch sun4u.

When I start the VM, I get a QEMU error: `qemu-sparc64-softmmu: Virtio VGA not available`

So far, I used UTM to run native VM's, not emulating other architectures.

The generated QEMU command is:
~~~
qemu-system-sparc64 -L /Applications/UTM.app/Contents/Resources/qemu -S \
    -spice unix=on,addr=E19FAA16-AC90-4092-910F-909FF610DAC2.spice,disable-ticketing=on,image-compression=off,playback-compression=off,streaming-video=off,gl=off \
    -chardev spiceport,id=org.qemu.monitor.qmp,name=org.qemu.monitor.qmp.0 \
    -mon chardev=org.qemu.monitor.qmp,mode=control \
    -net nic,model=lance,macaddr=DA:C2:67:E4:99:DE,netdev=net0 \
    -netdev vmnet-shared,id=net0 \
    -vga virtio-gpu-pci \
    -smp cpus=4,sockets=1,cores=4,threads=1 \
    -machine sun4u -accel tcg,tb-size=2048 -m 8192 \
    -audiodev spice,id=audio0 -device sb16,audiodev=audio0 \
    -device scsi-cd,bus=scsi.0,channel=0,scsi-id=0,drive=drive42A6C612-5ACE-49C8-8BE6-BE3DD960A0C7,bootindex=0 \
    -drive if=none,media=cdrom,id=drive42A6C612-5ACE-49C8-8BE6-BE3DD960A0C7,file=/Users/thierry/tmp/debian-12.0.0-sparc64-NETINST-1.iso,readonly=on \
    -device scsi-hd,bus=scsi.0,channel=0,scsi-id=1,drive=driveE698F144-12A0-4920-95B1-F0B57C512266,bootindex=1 \
    -drive if=none,media=disk,id=driveE698F144-12A0-4920-95B1-F0B57C512266,file=/Users/thierry/Library/Containers/com.utmapp.UTM/Data/Documents/vmsparc.utm/Data/E698F144-12A0-4920-95B1-F0B57C512266.qcow2,discard=unmap,detect-zeroes=unmap \
    -name vmsparc -uuid E19FAA16-AC90-4092-910F-909FF610DAC2
~~~

Any idea to fix this? (needless to say that Google did not help...)
Thanks

**Comment by adespoton** [comment]
To use the VirtIO video display, the guest OS needs a VirtIO video driver; you apparently don't have one on your install CD?

Switch displays to standard VGA or something else that the CD will have a driver for, and then play around with accelerated drivers once the base install is complete.

**Comment by lelegard** [comment]
Hi @adespoton 

> To use the VirtIO video display, the guest OS needs a VirtIO video driver; you apparently don't have one on your install CD?
> Switch displays to standard VGA or something else that the CD will have a driver for

And now I get a QEMU error: `qemu-sparc64-softmmu: unknown vga type: VGA`

Also tried all other display options from the UTM menu for the VM: `cirrus-vga`, `ati-vga`, `bochs-display`, `secondary-vga`, `vmware-svga` and all `virtio-*`.

None of the available display options in UTM work for SPARC64.
